UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Common/Helpers/NMHelper.cs

```diff
@@ -27,5 +28,57 @@ public static List<ShowFilesResponseModel> ListFilesResponse(BatchSearchResponse
             return ListshowFilesResponseModels;
         }
+
+        public static List<ShowDailyFilesResponseModel> GetDailyShowFilesResponse(BatchSearchResponse SearchResult)
+        {
+            List<ShowDailyFilesResponseModel> showDailyFilesResponses = new ();
+            List<AttributesModel> lstattributes = (SearchResult.Entries.Where(x => x.AllFilesZipSize.HasValue).Select(item => new AttributesModel
+            {
+                BatchId = item.BatchId,
+                DataDate = item.Attributes.Where(x => x.Key.Equals("Data Date")).Select(x => x.Value).FirstOrDefault(),
+                WeekNumber = item.Attributes.Where(x => x.Key.Equals("Week Number")).Select(x => x.Value).FirstOrDefault(),
+                Year = item.Attributes.Where(x => x.Key.Equals("Year")).Select(x => x.Value).FirstOrDefault(),
+                YearWeek = item.Attributes.Where(x => x.Key.Equals("Year / Week")).Select(x => x.Value).FirstOrDefault(),
+                AllFilesZipSize = (long)item.AllFilesZipSize
+            })).ToList();
+
+            IEnumerable<IGrouping<string, AttributesModel>> groupped = lstattributes.GroupBy(x => x.YearWeek);
+            foreach (IGrouping<string, AttributesModel> group in groupped)
+            {
+                List<DailyFilesDataModel> lstDataDate = (group.Select(item => new DailyFilesDataModel
+                {
+                    BatchId = item.BatchId,
+                    DataDate = item.DataDate,
+                    Filename = "Daily " + GetFormattedDate(item.DataDate) + ".zip",
+                    FileExtension = ".zip",
+                    FileDescription = "Daily " + GetFormattedDate(item.DataDate) + ".zip",
+                    AllFilesZipSize = item.AllFilesZipSize,
+                    FileSizeinKB = FileHelper.FormatSize(item.AllFilesZipSize),
+                    MimeType = "application/gzip"
+                })).Distinct().OrderBy(x => Convert.ToDateTime(x.DataDate)).ToList();
+
+                showDailyFilesResponses.Add(new ShowDailyFilesResponseModel
+                {
+                    YearWeek = group.Key,
+                    Year = group.Select(x => x.Year).FirstOrDefault(),
+                    WeekNumber = group.Select(x => x.WeekNumber).FirstOrDefault(),
+                    DailyFilesData = lstDataDate
+                });
+
+                showDailyFilesResponses = showDailyFilesResponses.OrderByDescending(x => x.Year).ThenByDescending(x => x.WeekNumber).ToList();
+            }
+            return showDailyFilesResponses;
+        }
+
+        public static string GetFormattedDate(string strdate)
+        {
+            string[] formats = { "M/d/yyyy", "d/M/yyyy", "M-d-yyyy", "d-M-yyyy", "d-MMM-yy", "d-MMMM-yyyy","yyyy-MM-dd" };
+            string retVal = "";
+
+            if (DateTime.TryParseExact(strdate, formats, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime date))
+                retVal = date.ToString("dd-MM-yy");
+
+            return retVal;
```

Comment:
Maybe better to simplify:
```suggestion
            if (DateTime.TryParseExact(strdate, formats, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime date))
                return date.ToString("dd-MM-yy");

            return string.Empty;
```

Reply:
code simplification done